### PR TITLE
format name for create schema, drop schema, etc.

### PR DESCRIPTION
**Pull request by Jimexist**
fixes https://github.com/prestodb/presto/issues/7465

### CODE REVIEW

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 1]
```diff
@@ -1746,6 +1747,33 @@ public void testQuantifiedComparison()
...
+        assertEquals(raw, formatted);
+    }
+
+    @Test
+    public void testQuotedSchemaRenaming() throws Exception
+    {
+        String raw = "alter schema \"123\" rename to \"1st\"";
+        Statement statement = SQL_PARSER.createStatement(raw);
+        String formatted = formatSql(statement, Optional.empty());
+        assertEquals(raw, formatted);
```
Comment (kokosing) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java]: you may want to wrap this block of code as a method and then just:
```
assertThatFormattedSqlEquals("alter...");
```

**Comment by Jimexist** [comment]
@kokosing squashed and refactored

**Comment by Jimexist** [comment]
@kokosing this fixes https://github.com/prestodb/presto/issues/7465 but i think a further PR is needed here because the case for table creation/deletion/alternation is the same.

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 2]
```diff
@@ -1746,6 +1746,27 @@ public void testQuantifiedComparison()
                 ));
     }
 
+    @Test
+    public void testQuotedSchemaCreation() throws Exception
+    {
+        String raw = "create schema \"123\"";
+        assertThatFormattedSqlEquals(raw);
```
Comment (electrum) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java]: The `assertStatement()` helper already handles this. Add this test to the existing `testCreateSchema()` method. Also, use a name that requires quoting, such as one containing a space:
```java
assertStatement("CREATE SCHEMA \"test abc\"",
        new CreateSchema(QualifiedName.of("test abc"), false, ImmutableMap.of()));
```

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 3]
```diff
@@ -1746,6 +1746,27 @@ public void testQuantifiedComparison()
...
+    }
+
+    @Test
+    public void testQuotedSchemaDrop() throws Exception
+    {
+        String raw = "drop schema \"123\"";
```
Comment (electrum) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java]: Same comments as above, please move to `testDropSchema()`

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 4]
```diff
@@ -1746,6 +1746,27 @@ public void testQuantifiedComparison()
...
+        assertThatFormattedSqlEquals(raw);
+    }
+
+    @Test
+    public void testQuotedSchemaRenaming() throws Exception
+    {
+        String raw = "alter schema \"123\" rename to \"1st\"";
```
Comment (electrum) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java]: Move to `testRenameSchema()`

**Comment by Jimexist** [comment]
@kokosing @electrum I took the liberty of adding more test cases.

But i think a better solution is to move escaping logic to `com.facebook.presto.sql.tree.QualifiedName#toString`. However this escalates the scope of this PR dramatically.

Looking forward to hearing your opinions.

**Comment by Jimexist** [comment]
the alternative is of course that I try to change all the use cases

**Comment by kokosing** [comment]
Please take a look travis failure.

The current solution is good to me. I would not move any formatting related logic to any `toString` method.

**Comment by Jimexist** [comment]
@kokosing seems travis is failing due to something i did not change - will rebuild after it is fixed

**Comment by Jimexist** [comment]
@electrum @kokosing CI passed

**Comment by Jimexist** [comment]
@electrum @kokosing any more comments?

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**